airflow/providers/http/hooks/http.py

```diff
@@ -126,15 +126,20 @@ def run(
         else:
             url = (self.base_url or '') + (endpoint or '')
 
+        request_parameters = dict(method=self.method, url=url, headers=headers)
+        request_parameters.update(request_kwargs)
+
         if self.method == 'GET':
             # GET uses params
-            req = requests.Request(self.method, url, params=data, headers=headers, **request_kwargs)
+            request_parameters["params"] = data
         elif self.method == 'HEAD':
             # HEAD doesn't use params
-            req = requests.Request(self.method, url, headers=headers, **request_kwargs)
+            pass
```

Comment:
Head requests can be used to get the content length of a download without actually downloading the data. I'm thinking we should not have `pass` here, what do you think?

Reply:
HEAD requests are still made, it's just they do no require data to be added into  payload, hence the `pass`.

Request instance is created later via `session.request` factory.